lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -976,6 +996,19 @@ defmodule Mix.Tasks.Xref do
         "cycles" ->
           {:cycles, print_cycles(file_references, filter, opts)}
 
+        "json" ->
+          {roots, callback, count} =
+            roots_and_callback(file_references, filter, sources, sinks, opts)
+
+          file_spec =
+            Mix.Utils.write_json_tree!("xref_graph.json", Enum.sort(roots), callback, opts)
+
+          if file_spec != "-" do
+            Mix.shell().info("Generated '#{file_spec}'.")
```

Comment:
Another nitpick:

```suggestion
            Mix.shell().info("Generated \"#{file_spec}\".")
```

:)